src/Functions/FunctionsHashing.h

```diff
@@ -896,17 +992,19 @@ class FunctionIntHash : public TargetSpecific::Default::FunctionIntHash<Impl, Na
 
 DECLARE_MULTITARGET_CODE(
 
-template <typename Impl>
+template <typename Impl, bool Keyed>
 class FunctionAnyHash : public IFunction
 {
+    static_assert(!Keyed || !Impl::use_int_hash_for_pods,
+                  "can't use IntHash does not support keys");
 public:
     static constexpr auto name = Impl::name;
 
 private:
     using ToType = typename Impl::ReturnType;
 
     template <typename FromType, bool first>
-    void executeIntType(const IColumn * column, typename ColumnVector<ToType>::Container & vec_to) const
+    void executeIntType(const IColumn * column, typename ColumnVector<ToType>::Container & vec_to, const ColumnWithTypeAndName & key) const
```

Comment:
Looks a bit weird and also is not consistent with the semantics in SQL to add another input argument after the output argument. Suggest order: 1. key 2. column 3. vec_to.